### ISSUE TITLE
Minesweeper: extra mine

### ISSUE THREAD

**Issue by AlexIljin** [factor/factor#1967]
Here's a screenshot, and I have two problems with it. First: it thinks there is an extra mine to be found somewhere, but I can't find an empty cell to put it. Second: the window size is a tad too small both vertically and horizontally.
![2018_03_05_20_58_39_minesweeper](https://user-images.githubusercontent.com/169039/36997014-90f0a37c-20b8-11e8-8d0c-f3cbab945c60.png)

**Comment by AlexIljin** [comment]
On the Hard level - how is it possible to place three mines here? There is only two empty cells, only one of which can have a mine. Previously it seemed that we are dealing with an "off-by-one" error, but here we are off by two:
![image](https://user-images.githubusercontent.com/169039/36997506-2320e828-20ba-11e8-984d-f1bdef73b86c.png)

**Comment by erg** [comment]
Try middle clicking on the numbers on the bottom row. I think there's one more row that isn't showing up.

**Comment by AlexIljin** [comment]
@erg: "I think there's one more row that isn't showing up."

I'm sure you are right, and I have a screenshot to prove it:
![image](https://user-images.githubusercontent.com/169039/36997787-0cce14b4-20bb-11e8-8d92-35ff4aeb3f5d.png)

**Comment by AlexIljin** [comment]
So, the mine counter is probably right, only the window size is wrong.

**Comment by mrjbq7** [comment]
The problem is ``resize-window`` is broken on Windows.  Minesweeper works just fine on macOS and Linux.

This button should do nothing (as it does on Linux and macOS), but it reduces the size of the window on Windows:

```factor
IN: scratchpad "Resize" [ drop world get { 200 200 } resize-window ] <button>
               T{ world-attributes { pref-dim { 200 200 } } } open-window
```

@erg

**Comment by mrjbq7** [comment]
Yes! I think I fixed it.

6ca7703831c75cc778a012e6250cc6175e698cf1

@AlexIljin can you test?

**Comment by AlexIljin** [comment]
@mrjbq7: are you saying that on Linux and macOS `resize-window` treats the `dim` parameter as the client area size?

**Comment by mrjbq7** [comment]
Yep, for example on macOS, we use a ``setContentSize:`` selector.

And on Linux we use ``gtk_window_resize`` which takes a content size.

**Comment by mrjbq7** [comment]
Any improvements are welcome! I'm just glad it's all consistent now.

**Comment by AlexIljin** [comment]
I've added the help for `resize-window`, but I'm too lazy to make a PR for just one commit.

**Comment by mrjbq7** [comment]
You can either paste the diff here as a patch and I'll apply it...  or put it in a branch and I can cherry-pick it if you give me the commit ref?

**Comment by mrjbq7** [comment]
Or include it in your next set of misc-fixes :-)

**Comment by AlexIljin** [comment]
All right, I like your idea. Here it is:
https://github.com/AlexIljin/factor/commit/de1e09ee8a527de7816b12c418badbd6ae0e1354

**Comment by mrjbq7** [comment]
Merged, thanks!
